### PR TITLE
update changelog for health checks

### PR DESCRIPTION
**Pull request by kschoche**
Updating the changelog for changes related to health checks, also did some minor formatting updates to a previous changelog entry.

### CODE REVIEW

File: CHANGELOG.md
Comment (kschoche) [inline CHANGELOG.md:4]: This whole section had 2 extra spaces so I've removed them.

File: CHANGELOG.md
Comment (ndhanushkodi) [inline CHANGELOG.md:7]: Maybe a comment on why it's recommended to enable TLS would be helpful, or a link to where in the doc it would describe that. So that I could know which network connections are insecure if I don't turn on TLS.

File: CHANGELOG.md
Comment (ndhanushkodi) [inline CHANGELOG.md:6]: This link doesn't resolve yet, but I'm assuming that's just coming in the future?

File: CHANGELOG.md
Comment (ndhanushkodi) [inline CHANGELOG.md:16]: The table in PR 368 was so helpful! The descriptions of the flags are clear and understandable. Great idea to link it here, hopefully something like that will make it into https://www.consul.io/docs/k8s/health.

**Review by ndhanushkodi — APPROVED** [review]
Awesome work!! I only had a few non-blocking comments.

File: CHANGELOG.md
Comment (kschoche) [reply CHANGELOG.md:6]: Good catch, there is an open PR here for it : https://github.com/hashicorp/consul/pull/8819 
I can hold off on merging this until it gets merged.

File: CHANGELOG.md
Comment (kschoche) [reply CHANGELOG.md:7]: doc in link ^ describes it

File: CHANGELOG.md
Comment (lkysow) [inline CHANGELOG.md:4]: 1. This needs a breaking change section as well. If any connect services were failing their kubernetes readiness checks previously then they were still routable through connect, now they won't be. Users should verify that their connect services are passing Kubernetes readiness probes prior to enabling health checks.
1. Can we add that in order to enable this feature for existing users they need to restart all the connect deployments so they are re-injected with a new label. Until they restart those deployments, health checks won't be synced.
1. I think we can be a bit more explicit here. Can we add something like "Now Kubernetes pods that fail their readiness checks will no longer receive Connect traffic until the readiness check passes"

File: CHANGELOG.md
Comment (lkysow) [reply CHANGELOG.md:7]: I think we should still document it here since it only requires a few more words ("because the health checks controller makes calls to Consul agents across all nodes so ACL tokens could be stolen if an attacker was watching the network")

File: CHANGELOG.md
Comment (lkysow) [inline CHANGELOG.md:10]: Let's be explicit about the bug that was fixed. Something like "Fix issue where Consul enterprise license job would fail for Consul versions >= 1.8.1".

Also we should link to the issue fixed for bug fixes if there is one (https://github.com/hashicorp/consul-helm/issues/647)

File: CHANGELOG.md
Comment (lkysow) [inline CHANGELOG.md:16]: These are consul-k8s deprecations. I don't think they should be listed for the consul-helm changelog since there's nothing for users to do.

What we do need to list is a breaking change that only consul-k8s v0.20.0 is supported with this helm chart version.

File: CHANGELOG.md
Comment (lkysow) [reply CHANGELOG.md:7]: Or your excellent description in values.yaml
```
  # NOTE: It is highly recommended to enable TLS with this feature because it requires
  # making calls to Consul clients across the cluster. Without TLS enabled, these calls
  # could leak ACL tokens.
```

**Review by thisisnotashwin — APPROVED** [review]
looks good!

**Pull request merged**
